### ISSUE TITLE
support function definition

### ISSUE THREAD

**Issue by allefeld** [allefeld/mkernel#2]
If a cell contains a function definition, i.e.

- the first line starts with `function`
- the last line contains only `end`, and
- the first line can be parsed as a function signature,

write the contents of the the cell to a file `name.m` in the current directory.

From 2024a, such notebook code remains usable if converted into a script, because Matlab now allows [local functions in scripts](https://uk.mathworks.com/help/matlab/matlab_prog/local-functions-in-scripts.html).

Or, in the conversion to a script these cells could be omitted, because presumably the notebook has been executed and the m-files been written.

Or, the m-files could be written to a temporary directory, which is added to the path, and which is deleted upon kernel cleanup.

The same could be done for class definitions, but Matlab does not allow local classes.

***

ChatGPT-generated code to extract the function name, or None:

```python
import re

def extract_matlab_function_name(code_string):
    """
    Extracts the MATLAB function name from a string.
    
    Parameters:
    code_string (str): A string containing MATLAB code.
    
    Returns:
    str: The name of the function if found, otherwise None.
    """
    # Regular expression to match MATLAB function definitions
    func_pattern = r'^\s*function\s+(?:\[[^\]]*\]\s*=|\s*[^\s=]+\s*=)?\s*(\w+)'
    
    # Search for the function name using the regex pattern
    match = re.search(func_pattern, code_string, re.MULTILINE)
    
    # Return the function name if found, otherwise return None
    if match:
        return match.group(1)
    else:
        return None
```

**Comment by allefeld** [comment]
See also IPython cell magic [`%%writefile`](https://ipython.readthedocs.io/en/stable/interactive/magics.html#cellmagic-writefile) and the metakernel cell magic [`%%file`](https://github.com/Calysto/metakernel/blob/main/metakernel/magics/README.md#file).

**Comment by allefeld** [comment]
Implemented by e625560, see <https://github.com/allefeld/mkernel/blob/main/README.md#functions>.
